solr/core/src/java/org/apache/solr/metrics/prometheus/core/SolrCoreMetric.java

```diff
@@ -30,18 +30,18 @@ public abstract class SolrCoreMetric extends SolrMetric {
   public String coreName;
 
   public SolrCoreMetric(Metric dropwizardMetric, String metricName) {
-    super(dropwizardMetric, metricName);
+    super(dropwizardMetric, metricName.substring(metricName.indexOf(".") + 1));
     Matcher cloudPattern = CLOUD_CORE_PATTERN.matcher(metricName);
     Matcher standalonePattern = STANDALONE_CORE_PATTERN.matcher(metricName);
     if (cloudPattern.find()) {
-      coreName = cloudPattern.group(1);
-      labels.put("core", cloudPattern.group(1));
-      labels.put("collection", cloudPattern.group(2));
-      labels.put("shard", cloudPattern.group(3));
-      labels.put("replica", cloudPattern.group(4));
+      coreName = cloudPattern.group("core");
+      labels.put("core", cloudPattern.group("core"));
+      labels.put("collection", cloudPattern.group("collection"));
+      labels.put("shard", cloudPattern.group("shard"));
+      labels.put("replica", cloudPattern.group("replica"));
     } else if (standalonePattern.find()) {
-      coreName = standalonePattern.group(1);
-      labels.put("core", standalonePattern.group(1));
+      coreName = standalonePattern.group("core");
+      labels.put("core", standalonePattern.group("core"));
```

Comment:
I meant for you to do this generically with a loop, thus without the code here actually referring to any names.  See `Matcher.namedGroups()`.